File: packet_seaquence/log/latest/work/src/protocol_analyzer/https.py
from src.protocol_analyzer.base import ProtocolAnalyzer

class AnalyzeHttps(ProtocolAnalyzer):
    """
    HTTPS内の詳細情報を解析するクラス
    """

    def analyze(self):
        # HTTPSは暗号化されるため、詳細フィールドを取得できない場合が多い
        if not hasattr(self.packet, 'ssl') and not hasattr(self.packet, 'tls'):
            return None
        try:
            https_info = {}
            if hasattr(self.packet, 'ssl'):
                https_info['ssl_record'] = True
            if hasattr(self.packet, 'tls'):
                https_info['tls_record'] = True
            return https_info
        except Exception as e:
            print(f"HTTPS解析中にエラーが発生しました: {str(e)}")
            return None

    def get_display_info(self):
        info = self.analyze()
        if not info:
            return "HTTPS情報なし"
        return "HTTPS セッション (暗号化)"

    def get_summary(self):
        info = self.analyze()
        if not info:
            return "HTTPS"
        return "HTTPS Session"

File: packet_seaquence/log/go_src/protocol_analyzer/https.go
// Package protocol_analyzer provides HTTPS protocol analyzer
package protocol_analyzer

// AnalyzeHTTPS implements the ProtocolAnalyzer interface for HTTPS
type AnalyzeHTTPS struct {
	Pkt Packet
}

// NewAnalyzeHTTPS creates a new HTTPS analyzer
func NewAnalyzeHTTPS(packet Packet) *AnalyzeHTTPS {
	return &AnalyzeHTTPS{
		Pkt: packet,
	}
}

// Analyze extracts HTTPS information from a packet
// HTTPS packets are encrypted, so detailed information may not be available
func (a *AnalyzeHTTPS) Analyze() map[string]interface{} {
	if !a.Pkt.LayerExists("ssl") && !a.Pkt.LayerExists("tls") {
		return nil
	}

	httpsInfo := make(map[string]interface{})
	
	if a.Pkt.LayerExists("ssl") {
		httpsInfo["ssl_record"] = true
	}
	
	if a.Pkt.LayerExists("tls") {
		httpsInfo["tls_record"] = true
	}
	
	return httpsInfo
}

// GetDisplayInfo returns formatted HTTPS information
func (a *AnalyzeHTTPS) GetDisplayInfo() string {
	info := a.Analyze()
	if info == nil {
		return "HTTPS情報なし"
	}
	
	return "HTTPS セッション (暗号化)"
}

// GetSummary returns a summary of HTTPS information
func (a *AnalyzeHTTPS) GetSummary() string {
	info := a.Analyze()
	if info == nil {
		return "HTTPS"
	}
	
	return "HTTPS Session"
}